docs/understand/cmake_packages.rst

```diff
@@ -1,139 +1,188 @@
+driving the compilation of HIP programs on Nvidia runtimes. As such, when
+dependencies are not found in standard system locations, one either has to
+instruct CMake to search for package config files in additional folders using
+the ``CMAKE_PREFIX_PATH`` variable (a semi-colon separated list of filesystem
+paths), or using ``<PackageName>_ROOT`` variable on a project-specific basis.
+
+There are nearly a dozen ways to set these variables. One may be more convenient
+over the other depending on your workflow. Conceptually the simplest is adding
+it to your CMake configuration command on the command-line via
+``-D CMAKE_PREFIX_PATH=....`` . AMD packaged ROCm installs can typically be
+added to the config file search paths such as:
+
+-  Windows: ``-D CMAKE_PREFIX_PATH=${env:HIP_PATH}``
+
+-  Linux: ``-D CMAKE_PREFIX_PATH=/opt/rocm``
 
 ROCm provides the respective *config-file* packages, and this enables
 ``find_package`` to be used directly. ROCm does not require any Find module as
 the *config-file* packages are shipped with the upstream projects.
 
-Finding Dependencies
---------------------
+For a complete guide on where and how ROCm may be installed on a system, refer
+to the installation guides in these docs (`Linux <../deploy/linux/index.html>`_).
+
+Using HIP in CMake
+==================
 
-When dependencies are not found in standard locations such as */usr* or
-*/usr/local*, then the ``CMAKE_PREFIX_PATH`` variable can be set to the
-installation prefixes. This can be set to multiple locations with a semicolon
-separating the entries.
+ROCm componenents providing a C/C++ interface support being consumed using any
+C/C++ toolchain that CMake knows how to drive. ROCm also supports CMake's HIP
+language features, allowing users to program using the HIP single-source
+programming model. The HIP API without compiling GPU device code behaves as a
+C/C++ library.
 
-There are two ways to set this variable:
+Using the HIP single-source programming model
+---------------------------------------------
 
--  Pass the flag when configuring with ``-DCMAKE_PREFIX_PATH=....`` This
-   approach is preferred when users install the components in custom locations.
+Source code written in the HIP dialect of C++ typically use the `.hip`
+extension. When the HIP CMake language is enabled, it will automatically
+associate such source files with the HIP toolchain being used. Because HIP
+itself relies on a C++ Standard Template Library (among other C++ toolchain
+fragments), enabling the C++ language is also recommended.
 
--  Append the variable in the CMakeLists.txt file. This is useful if the
-   dependencies are found in a common location. For example, when the binaries
-   provided on `repo.radeon.com <http://repo.radeon.com>`_ are installed to
-   */opt/rocm*, you can add the following line to a CMakeLists.txt file
+::
 
-   ::
+    cmake_minimum_required(VERSION 3.21) # enable_language(HIP)
```

Comment:
I think the comment here is trying to say "you need 3.21 to enable the HIP language"?

If it's instead trying to show that HIP can also be enabled by `enable_language`, I would put the comment next to the `project` call saying "# or, the user can call `enable_language(HIP)` directly"

Reply:
It should also be `3.21.3`.

Reply:
3.21.3 added.